File: Leetcode/DeleteCharacterstoMakeFancyString.rs
// https://leetcode.com/problems/delete-characters-to-make-fancy-string

struct Solution {}

impl Solution {
    pub fn make_fancy_string(s: String) -> String {
        let s = s.into_bytes();
        let mut a = vec![];
        a.extend(s.iter().take(2));
        for c in s.windows(3) {
            if c[0] == c[1] && c[1] == c[2] {
            } else {
                a.push(c[2]);
            }
        }
        String::from_utf8(a).unwrap()
    }
}

File: Leetcode/DeleteCharacterstoMakeFancyString.cpp
// https://leetcode.com/problems/delete-characters-to-make-fancy-string

class Solution {
   public:
    string makeFancyString(string s) {
        string ans;

        for (size_t i = 0; i < s.size(); ++i) {
            const auto sz = ans.size();
            if (sz > 1) {
                if (ans[sz - 2] == ans[sz - 1] && ans[sz - 1] == s[i]) {
                    // skip
                } else {
                    ans.push_back(s[i]);
                }
            } else {
                ans.push_back(s[i]);
            }
        }

        return ans;
    }
};